pkg/services/contexthandler/contexthandler.go

```diff
@@ -110,6 +111,35 @@ func FromContext(c context.Context) *contextmodel.ReqContext {
 	return nil
 }
 
+// CopyWithReqContext returns a copy of the parent context with a cloned version of its RequestContext value
+func CopyReqContext(ctx context.Context) context.Context {
+	origReqCtx := FromContext(ctx)
+	if origReqCtx == nil {
+		return ctx
+	}
+
+	webCtx := &web.Context{
+		Req:  origReqCtx.Req.Clone(ctx),
+		Resp: web.NewResponseWriter(origReqCtx.Req.Method, response.CreateNormalResponse(http.Header{}, []byte{}, 0)),
+	}
+	reqCtx := &contextmodel.ReqContext{
+		Context:               webCtx,
+		SignedInUser:          origReqCtx.SignedInUser,
```

Comment:
Nit. The fields that are pointers are not really cloned. Not sure we need to care of this, but something to keep in mind since if modified and read by multiple goroutines it would panic. Think it would be good if the comment of the function explains this clearly.

Reply:
Added better docs here: [9aea1b1](https://github.com/grafana/grafana/pull/70100/commits/9aea1b1812c197a8e6a406a14bb64e1559d2daab)